### PR TITLE
Fix a bug with array types and StokesProfile

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

File: gwcs/coordinate_frames.py [hunk 1]
```diff
@@ -731,8 +731,8 @@ def from_index(cls, indexes):
         """
 
         nans = np.isnan(indexes)
-        indexes = np.asanyarray(indexes, dtype=int)
-        out = np.empty_like(indexes, dtype=object)
+        indexes = np.asarray(indexes, dtype=int)
```
Comment (Cadair) [inline gwcs/coordinate_frames.py:734]: The bug here was if indexes was a Quantity then it created a quantity for the strings which didn't work.

**Comment by Cadair** [comment]
@nden This is blocking some critical fixes to my DKIST work in preparation for us starting to receive data next month if you could get it out into the world as soon as possible that would be great. Thanks.

**Comment by nden** [comment]
@Cadair I can make a quick release. Could you add a test for this? Thanks!

**Comment by Cadair** [comment]
I did? Or did I miss something?

**Comment by nden** [comment]
Ah yes, I missed it.

**Review by nden — APPROVED** [review]

**Comment by nden** [comment]
@Cadair gwcs 0.17.1 is on PyPi.

**Comment by Cadair** [comment]
Thanks so much!

**Pull request merged**